crates/ty_python_semantic/resources/mdtest/snapshots/final.md_-_Tests_for_the_`@typi…_-_Cannot_override_a_me…_(338615109711a91b).snap

```diff
@@ -0,0 +1,414 @@
+14 |     @final
+15 |     @classmethod
+16 |     def class_method1(cls) -> int: ...
+17 |     @staticmethod
+18 |     @final
+19 |     def static_method1() -> int: ...
+20 |     @final
+21 |     @classmethod
+22 |     def class_method2(cls) -> int: ...
+23 |     @staticmethod
+24 |     @final
+25 |     def static_method2() -> int: ...
+26 |     @lossy_decorator
+27 |     @final
+28 |     def decorated_1(self): ...
+29 |     @final
+30 |     @lossy_decorator
+31 |     def decorated_2(self): ...
+32 | 
+33 | class Child(Parent):
+34 |     def foo(self): ...  # error: [override-of-final-method]
+35 |     @property
+36 |     def my_property1(self) -> int: ...  # error: [override-of-final-method]
+37 |     @property
+38 |     def my_property2(self) -> int: ...  # error: [override-of-final-method]
+39 |     @classmethod
+40 |     def class_method1(cls) -> int: ...  # error: [override-of-final-method]
+41 |     @staticmethod
+42 |     def static_method1() -> int: ...  # error: [override-of-final-method]
+43 |     @classmethod
+44 |     def class_method2(cls) -> int: ...  # error: [override-of-final-method]
+45 |     @staticmethod
+46 |     def static_method2() -> int: ...  # error: [override-of-final-method]
+47 |     def decorated_1(self): ...  # TODO: should emit [override-of-final-method]
+48 |     @lossy_decorator
+49 |     def decorated_2(self): ...  # TODO: should emit [override-of-final-method]
+50 | 
+51 | class OtherChild(Parent): ...
+52 | 
+53 | class Grandchild(OtherChild):
+54 |     @staticmethod
+55 |     # TODO: we should emit a Liskov violation here too
+56 |     # error: [override-of-final-method]
+57 |     def foo(): ...
+58 |     @property
+59 |     # TODO: we should emit a Liskov violation here too
+60 |     # error: [override-of-final-method]
+61 |     def my_property1(self) -> str: ...
+62 |     # TODO: we should emit a Liskov violation here too
+63 |     # error: [override-of-final-method]
+64 |     class_method1 = None
+65 | 
+66 | # Diagnostic edge case: `final` is very far away from the method definition in the source code:
+67 | 
+68 | T = TypeVar("T")
+69 | 
+70 | def identity(x: T) -> T: ...
+71 | 
+72 | class Foo:
+73 |     @final
+74 |     @identity
+75 |     @identity
+76 |     @identity
+77 |     @identity
+78 |     @identity
+79 |     @identity
+80 |     @identity
+81 |     @identity
+82 |     @identity
+83 |     @identity
+84 |     @identity
+85 |     @identity
+86 |     @identity
+87 |     @identity
+88 |     @identity
+89 |     @identity
+90 |     @identity
+91 |     @identity
+92 |     def bar(self): ...
+93 | 
+94 | class Baz(Foo):
+95 |     def bar(self): ...  # error: [override-of-final-method]
+```
+
+# Diagnostics
+
+```
+error[override-of-final-method]: Cannot override `Parent.foo`
+  --> src/mdtest_snippet.pyi:34:9
+   |
+33 | class Child(Parent):
+34 |     def foo(self): ...  # error: [override-of-final-method]
+   |         ^^^ Override of `foo` from superclass `Parent`
```

Comment:
Should we omit `foo` here, given that it already appears in the primary message (and the name is also fairly obvious from what's underlined).

Overrides the definition from its superclass `Parent`